### PR TITLE
feat: html summarized report

### PR DESCRIPTION
**Pull request by tejugang**
# Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] Optimization

# Description  
- Added build_chaos_report_html() function in a new krkn/summarized_reports/transform_html.py module that generates a self-contained, interactive HTML report mirroring the existing PDF report section-by-section
- Added generate_html_report config parameter (defaults to True) independent of the existing generate_pdf_report flag
- Wired up HTML report generation in run_kraken.py alongside the PDF output
- Added generate-html-report entry to containers/krknctl-input.json for container/krknctl support

## Related Tickets & Documents
If no related issue, please create one and start the converasation on wants of 

- Related Issue #: #1519 
- Closes #: 

# Documentation  
- [x] **Is documentation needed for this update?**

If checked, a documentation PR must be created and merged in the [website repository](https://github.com/krkn-chaos/website/).

## Related Documentation PR (if applicable)  
[documentation PR](https://github.com/krkn-chaos/website/pull/592)

# Checklist before requesting a review
- [x] Ensure the changes and proposed solution have been discussed in the relevant issue and have received acknowledgment from the community or maintainers. See [contributing guidelines](https://krkn-chaos.dev/docs/contribution-guidelines/)
See [testing your changes](https://krkn-chaos.dev/docs/developers-guide/testing-changes/) and run on any Kubernetes or OpenShift cluster to validate your changes
- [x] I have performed a self-review of my code by running krkn and specific scenario 
- [ ] If it is a core feature, I have added thorough unit tests with above 80% coverage

*REQUIRED*:
Description of combination of tests performed and output of run

```bash
python run_kraken.py
...
================================================================================
KRKN RUN SUMMARY
================================================================================
Run UUID : 21749365-f8c7-4fc9-865a-0799486644d6
Status   : FAIL
Cluster  : 4.22.0 (AWS, self-managed)
Window   : 2026-08-12 15:14:03 – 15:14:03
Nodes    : 6
Network  : OVNKubernetes
CLUSTER OVERVIEW
  Type     Count  Instance       Arch    Kubelet    OS
  worker   3      m6i.xlarge     amd64   v1.35.5    Red Hat Enterprise Linux CoreOS 9.8.20260520-0 (Plow)
  master   3      m6i.xlarge     amd64   v1.35.5    Red Hat Enterprise Linux CoreOS 9.8.20260520-0 (Plow)
TARGETS
  [1] Scenario  : scenarios/openshift/network_chaos_ingress.yml (pod_network_scenarios)
  Label Selector : <label_selector>
KEY METRICS
  [1] Scenario: scenarios/openshift/network_chaos_ingress.yml
    Exit Status           : FAIL (1)
ALERTS & SLOs
  SLOs Evaluated  : 27
  SLOs Passed     : 26 / 27
  SLOs Failed     : 1
  Critical Alerts : None
  Error Logs      : 1
    [2026-08-12T19:14:03.097Z] NativeScenarioPlugin exiting due to Exception Validation failed for 'NetworkScenarioConfig': Invalid parameter 'kraken_config', expected one of: node_interface_name, image, label_selector, test_duration, wait_duration, instance_count, kubeconfig_path, execution_type, network_params
FAILED SLOs (per scenario)
  Scenario: scenarios/openshift/network_chaos_ingress.yml
    FAIL  [warning ]  etcd database size in use is less than 50% of the actual allocated storage.
RESILIENCY SCORE
  scenarios/openshift/network_chaos_ingress.yml : 97 / 100
  Overall Score                : 97 / 100  ✅
================================================================================
2026-08-12 15:14:13,511 [INFO] PDF report generated: /Users/tgangise/krkn/kraken.report.pdf

file:///Users/tgangise/krkn/kraken.report.pdf

2026-08-12 15:14:13,512 [INFO] HTML report generated: /Users/tgangise/krkn/kraken.report.html
file:///Users/tgangise/krkn/kraken.report.html

2026-08-12 15:14:13,512 [INFO] api_url not set, skipping telemetry upload.
2026-08-12 15:14:13,512 [INFO] Kraken UUID for the run: 21749365-f8c7-4fc9-865a-0799486644d6. Report generated at kraken.report.
2026-08-12 15:14:13,512 [ERROR] Post scenarios are still failing at the end of all iterations
```

### CODE REVIEW

**Comment by paigerube14** [comment]
let's be sure that the html file gets archived in the CI runs

File: krkn/summarized_reports/transform_html.py [hunk 1]
```diff
@@ -0,0 +1,659 @@
...
+def _html_section(title, body):
+    return (
+        f'<div class="section">'
+        f'<div class="section-header">{_h(title)}</div>'
+        f'<div class="section-body">{body}</div>'
+        f"</div>"
+    )
+
+
+def build_chaos_report_html(chaos_output: dict, output_path: str) -> str:
```
Comment (ddjain) [inline krkn/summarized_reports/transform_html.py:140]: Could you please add unit tests for build_chaos_report_html() covering basic output validation and edge cases (e.g., empty scenarios, missing fields) to help safeguard against regressions?

File: config/config.yaml [hunk 2]
```diff
@@ -8,6 +8,7 @@ kraken:
     signal_address: 0.0.0.0                                # Signal listening address
     port: 8081                                             # Signal port
     generate_pdf_report: True                              # Generate a PDF summary report after the run
+    generate_html_report: True                              # Generate an interactive HTML summary report after the run
```
Comment (chaitanyaenr) [inline config/config.yaml]: @tejugang Thoughts on switching to the following format to make it more scalable:
```
report_formats:
  - pdf
  - html
 ```

Comment (paigerube14) [reply config/config.yaml]: agree, this might be better

**Comment by paigerube14** [comment]
the html output in the functional tests look great, one small nit I agree with Ravi on with the output types listing pdf and html. [And this small print issue ](https://github.com/krkn-chaos/krkn/pull/1555/changes#r3769692249)

File: run_kraken.py [hunk 3]
```diff
@@ -105,7 +106,13 @@ def main(options, command: Optional[str], out: Optional[dict] = None) -> int:
             config["kraken"], "publish_kraken_status", False
         )
         port = get_yaml_item_value(config["kraken"], "port", 8081)
-        generate_pdf_report = get_yaml_item_value(config["kraken"], "generate_pdf_report", True)
+        report_formats = get_yaml_item_value(config["kraken"], "report_formats", None)
+        if report_formats is not None:
+            generate_pdf_report = "pdf" in report_formats
+            generate_html_report = "html" in report_formats
+        else:
+            generate_pdf_report = get_yaml_item_value(config["kraken"], "generate_pdf_report", True)
```
Comment (paigerube14) [inline run_kraken.py]: is this variable correct? shouldn't we use report_formats? or is this a backup based on previous PR

Comment (tejugang) [reply run_kraken.py]: Yes, this is kept as a fallback for backward compatibility but I updated Krkn-hub to use report_formats as the new parameter here: https://github.com/krkn-chaos/krkn-hub/pull/377
